sites/game-client/ui/main-ui.mjs

```diff
@@ -43,28 +54,80 @@ class MainUI {
    * 
    * @param target The element which wants add button, should be an Overlay or Modal.
    */
-  addToolbarButton(target)
+  addToolbarButton(target) {
+    // TODO(fanlan1210)
+  }
 
   /**
    * Remove an button from the Toolbar.
    * 
    * @param target The element which wants remove button, should be an Overlay or Modal.
    */
-  removeToolbarButton(target)
+  removeToolbarButton(target) {
+    // TODO(fanlan1210)
+  }
 
   // TODO: Stack visualization? Stack height limit?
   /**
    * Show notification in notification area.
    * @param msg The message want to be shown.
    * @param {Number} timeout The duraion of showing notification, the unit is millisecond.
    */
-  showNotification(msg, timeout);
+  showNotification(msg, timeout) {
+    // TODO(lisasasasa)
+  }
 
   // If we want to use individual overlay for meeting
   // position = 0...n
   // setTopbarOverlay(position, overlay);
   // count=0 for no topbar.
   // resizeTopbarOverlay(count);
+
+  // ========== Internal Methods ============
+  // Methods below should only be used by UI classes such as Model/Overlay.
+
+  /**
+   * Send a DOM element back into staging area.
+   */
+  restoreToStaging(dom) {
+    document.getElementById(STAGING_ID).appendChild(dom);
```

Comment:
If `STAGING_ID` is a constant, consider caching the DOM element instead of querying it every time.